std/binary.asm

```diff
@@ -15,15 +16,20 @@ machine Binary(latch, operation_id) {
     col fixed latch(i) { if (i % 4) == 3 { 1 } else { 0 } };
     col fixed FACTOR(i) { 1 << (((i + 1) % 4) * 8) };
 
-    col fixed P_A(i) { i % 256 };
-    col fixed P_B(i) { (i >> 8) % 256 };
-    col fixed P_operation(i) { (i / (256 * 256)) % 3 };
+    // TOOD would be nice with destructuring assignment for arrays.
+    let inputs: (int -> int)[] = split_bits([2, 8, 8]);
```

Comment:
```suggestion
    let input_output_pairs: (int -> int)[] = cross_product([3, 256, 256]);
```

What do you think of this? Two changes:
- I'm thinking of it in terms of a cross product, so at least for me it would be a better name.
- I think the function could be rewritten to support cross products of ranges of arbitrary size. This would actually yield a different result and enforce that `0 <= op < 3`. Then the match below would actually be exhaustive, like it was before.

Reply:
sounds good!

Reply:
We could even have a type at some point that stores a function and its cycle size. I think that is useful for implementing the `[....]*`-opreator as well.